### PR TITLE
- listing item with ngx-gallery 

### PR DESCRIPTION
**Pull request by cis-shubham-t**

### CODE REVIEW

**Comment by AllienWorks** [comment]
WIP, please don't merge yet.

I'm redesigning the page a bit :)

**Comment by AllienWorks** [comment]
I'm done with my changes. I got a bit carried away, redesigning the whole Listing's detail.

screenshot from 2018-11-01 13-30-53

In the pricing section below item's description, there's now more detailed info about shipping/escrow etc.

### What needs to be finished

All marked with `TO_DO` class and our classic pink shadow (see screen ^):
- display correct escrow value for local buyers
- sum "total amount needed for order" for both worldwide & local buyers (item price + shipping + escrow)

**Comment by cis-shubham-t** [comment]
hello @AllienWorks 
I'm done with my changes. please have a look over it. and please late me know if I miss something?
image-preview

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 1]
```diff
@@ -10,97 +10,204 @@
...
-        <div class="small image" fxFlex="1 1 100px" *ngFor="let img of data?.listing?.imageCollection?.images" (click)="data?.listing?.imageCollection.setPreviewImage(img)">
-          <img src="{{img.thumbnail}}">
+        <div class="item price">
+          <div class="title">Price</div>
+          <div class="value part">
+            <span class="big">{{ data?.listing?.basePrice.getIntegerPart() }}</span><span class="big">{{ data?.listing?.basePrice.dot() }}</span><span class="small">{{ data?.listing?.basePrice.getFractionalPart() }}</span>
+            <span class="currency">PART</span>
+          </div>
+          <div class="value fiat">
+            <span class="fiat">~ {{price?.usd}} xx EUR</span>
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html:46]: Is this the USD or EUR price?

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 2]
```diff
@@ -10,97 +10,204 @@
...
+        </thead>
+        <tbody>
+          <tr>
+            <th><strong>Shipping</strong></th>
+            <td>
+              <span class="part price">
+                <strong>{{ data?.listing?.internationalShippingPrice.getAmount() }}</strong> <small class="currency">PART</small>
+              </span>
+              <span class="fiat price">
+                &asymp; 0.41 <small class="currency">EUR</small>
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html:98]: This is hard-coded?

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 3]
```diff
@@ -10,97 +10,204 @@
...
+              </span>
+            </td>
+            <td>
+              <span class="part price">
+                <strong>{{ data?.listing?.domesticShippingPrice.getAmount() }}</strong> <small class="currency">PART</small>
+              </span>
+              <span class="fiat price">
+                &asymp; 0.41 <small class="currency">EUR</small>
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html:106]: This is hard-coded?

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 4]
```diff
@@ -10,97 +10,204 @@
...
+            <th>
+              <strong>Escrow</strong><br>
+              (refunded upon delivery)
+            </th>
+            <td>
+              <span class="part price">
+                <strong>{{ data?.listing?.escrowPriceInternational?.getAmount() }}</strong> <small class="currency">PART</small>
+              </span>
+              <span class="fiat price">
+                &asymp; 0.41 <small class="currency">EUR</small>
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html:120]: This is hard-coded?

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 5]
```diff
@@ -10,97 +10,204 @@
...
+              </span>
+            </td>
+            <td>
+              <!-- FIXME: "escrow for local (not international)" needs to be implemented: -->
+              <span class="part price">
+                <strong>{{ data?.listing?.escrowPriceDomestic?.getAmount() }}</strong> <small class="currency">PART</small>
+              </span>
+              <span class="fiat price">
+                &asymp; 0.41 <small class="currency">EUR</small>
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html:129]: This is hard-coded?

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 6]
```diff
@@ -10,97 +10,204 @@
...
+              <strong>Total needed for order</strong><br>
+              (incl. refundable Escrow)
+            </th>
+            <!-- TODO: price + shipping + escrow -->
+            <td>
+              <span class="part price">
+                <strong>{{data?.listing?.totalAmountInternaltional?.getAmount()}}</strong> <small class="currency">PART</small>
+              </span>
+              <span class="fiat price">
+                &asymp; 0.41 <small class="currency">EUR</small>
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html:144]: This is hard-coded?

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 7]
```diff
@@ -10,97 +10,204 @@
...
+              </span>
+            </td>
+            <td>
+              <span class="part price">
+                <strong>{{data?.listing?.totalAmountDomestic?.getAmount()}}</strong> <small class="currency">PART</small>
+              </span>
+              <span class="fiat price">
+                &asymp; 0.41 <small class="currency">EUR</small>
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html:152]: This is hard-coded?

**Review by zaSmilingIdiot — APPROVED** [review]
I'm not too thrilled about the inclusion of things that are not implemented yet. Specifically because the implementation of the items has no future implementation plan as yet.

In saying that though, the not-implemented-as-yet functionality is currently hidden, and so I'll ignore its inclusion for the sake of completing the PR for testing. Lets keep in mind that for future work, we either need a ticket to exist for that functionality so that we do actually get to implementing it at some point, or otherwise leave the functionality out of the implementation until its meant to be included (it introduces future code debt and/or future possible bugs when functionality changes).

**Pull request merged**